src/extension_instance.ml

```diff
@@ -46,10 +46,24 @@ end = struct
     in
     let (lazy outputChannel) = Output.language_server_output_channel in
     let revealOutputChannelOn = LanguageClient.RevealOutputChannelOn.Never in
+    let initializationOptions =
+      let codelens =
+        Option.map
+          Settings.(get server_codelens_setting)
+          ~f:(fun enable -> LanguageClient.OcamllspSetting.create ~enable ())
+      in
+      let extendedHover =
+        Option.map
+          Settings.(get server_extendedHover_setting)
+          ~f:(fun enable -> LanguageClient.OcamllspSetting.create ~enable ())
+      in
+      LanguageClient.OcamllspConfig.create ?codelens ?extendedHover ()
```

Comment:
I think that this format is correct for the initialization options, but I'm not sure how to confirm

Reply:
initializationOptions and configuration are not the same thing. See this discussion https://github.com/Microsoft/language-server-protocol/issues/567

Reply:
apparently using `workspace/didChangeConfiguration` with a payload is also bad, so what is recommended in ocamllsp is not good practice.

https://github.com/microsoft/language-server-protocol/issues/567#issuecomment-448538082

Looks like they never actually updated the spec with something clearer :(

Reply:
Indeed. Feel free to update the server to account for this.

Reply:
I don't have too much time to dedicate to this issue currently. So for now I dropped the `initializationOptions` part and made sure the `workspace/didChangeConfiguration` way works, which should give good support of ocamllsp 1.16.{1,2}.

I created an issue and I'll comment there if I have time to work on the switch to `workspace/configuration`
https://github.com/ocaml/ocaml-lsp/issues/1160